src/scripts/main.js

```diff
@@ -1,3 +1,20 @@
 'use strict';
 
-// write your code here
+const table = document.querySelector('table');
+
+function cloneAndInsertColumn(columnIndex) {
+  const rows = table.querySelectorAll('tr');
+
+  rows.forEach((row) => {
+    const cells = row.children;
+    const targetCell = cells[columnIndex];
+
+    if (targetCell) {
+      const clonedCell = targetCell.cloneNode(true);
+
+      row.insertBefore(clonedCell, cells[cells.length - 1]);
```

Comment:
The cloned cell is being inserted before the last cell of each row. If the intention is to duplicate the column at the same position, you should insert the cloned cell at `cells[columnIndex + 1]` instead of `cells[cells.length - 1]`.